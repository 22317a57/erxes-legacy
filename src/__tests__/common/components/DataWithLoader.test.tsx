import { mount, shallow } from 'enzyme';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import DataWithLoader from '../../../modules/common/components/DataWithLoader';

describe('DataWithLoader component', () => {
  const defaultProps = {
    data: 'today',
    loading: false
  };

  test('renders DataWithLoader successfully', () => {
    shallow(<DataWithLoader {...defaultProps} />);
  });

  test('snapshot matches', () => {
    const rendered = renderer
      .create(<DataWithLoader {...defaultProps} />)
      .toJSON();

    expect(rendered).toMatchSnapshot();
  });

  test('renders successfully with default value', () => {
    const wrapper = mount(<DataWithLoader {...defaultProps} />);
    const props = wrapper.props();

    expect(props).toMatchObject(defaultProps);
  });
});
