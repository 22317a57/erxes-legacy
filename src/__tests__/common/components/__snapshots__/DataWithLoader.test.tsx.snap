// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataWithLoader component snapshot matches 1`] = `"today"`;
